src/CacheAdapters/Memory.cpp

```diff
@@ -0,0 +1,59 @@
+#include "Common.h"
+#include "CacheAdapters/Memory.h"
+#include "SSEConfig.h"
+#include "SSEEvent.h"
+
+using namespace std;
+
+Memory::Memory(int length) {
+    _config.length = length;
+}
+
+void CacheInterface::CacheEvent(SSEEvent* event) {
+  // If we have the event id in our vector already don't remove it.
+  // We want to keep the order even if we get an update on the event.
+  if (std::find(_cache_keys.begin(), _cache_keys.end(), event->getid()) == _cache_keys.end()) {
+    _cache_keys.push_back(event->getid());
+  }
+
+  _cache_data[event->getid()] = SSEEventPtr(event);
+
+  // Delete the oldest cache object if we hit the historyLength limit.
+  if ((int)_cache_keys.size() > _config.length) {
+    string &firstElementId = *(_cache_keys.begin());
+    _cache_data.erase(firstElementId);
```

Comment:
I'm not sure if it matters and a race condition is possible in this instance.

Should the key maybe be deleted before the data? Could a situation arise where the cache data is deleted and before the key gets deleted another thread tries to read data using that key?